### PR TITLE
feat(core): add ConsentStatus to UsersPageQuery

### PR DESCRIPTION
**Pull request by xflord**
* added consent status to UsersPageQuery to be able to find users with consents in set state in facility

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java [hunk 1]
```diff
@@ -1620,6 +1622,15 @@ private String getSQLJoinFacility(UsersPageQuery query, MapSqlParameterSource na
...
 				joinString += " join resource_services on resources.id=resource_services.resource_id and resource_services.service_id=(:serviceId)";
 			}
+
+			if (query.getConsentStatuses() != null && !query.getConsentStatuses().isEmpty()) {
+				List<String> statusStrings = query.getConsentStatuses().stream()
+					.map(ConsentStatus::toString).toList();
+				namedParams.addValue("consentStatuses", statusStrings);
+				joinString += " join consents on users.id = consents.user_id and (select consents.status from consents where consents.user_id=users.id" +
+					" and consents.consent_hub_id in (select consent_hub_id from consent_hubs_facilities where consent_hubs_facilities.facility_id=(:facilityId)) order by consents.modified_at desc limit 1)::text" +
+					" in (:consentStatuses) and consents.consent_hub_id in (select consent_hub_id from consent_hubs_facilities where consent_hubs_facilities.facility_id=(:facilityId))";
```
[the remark this answers is not in the document]
Comment (xflord) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java:1632]: Without the AND part the query would return consents from any consent hub, as long as its status was same as the one from the status matching part, no?

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java [hunk 2]
```diff
@@ -1620,6 +1622,15 @@ private String getSQLJoinFacility(UsersPageQuery query, MapSqlParameterSource na
 				namedParams.addValue("serviceId", query.getServiceId());
 				joinString += " join resource_services on resources.id=resource_services.resource_id and resource_services.service_id=(:serviceId)";
 			}
+
+			if (query.getConsentStatuses() != null && !query.getConsentStatuses().isEmpty()) {
+				List<String> statusStrings = query.getConsentStatuses().stream()
+					.map(ConsentStatus::toString).toList();
+				namedParams.addValue("consentStatuses", statusStrings);
+				joinString += " join consents on users.id = consents.user_id and (select consents.status from consents where consents.user_id=users.id" +
```
Comment (cuadradek) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java:1630]: Just a nitpick: this SQL query is quite complicated, and it is hard to read. Could you please reformat the query, so it looks something like this? 
image

Comment (xflord) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java:1630]: You're right, should have done this initially. My bad

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Johaney-s) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java]: you're right. i tried to simplify the query somehow, but only managed to get something more readable (to me), but extremely slow, so i'm happy now with your solution :smile:

**Review by Johaney-s — APPROVED** [review]

**Review by cuadradek — APPROVED** [review]

**Pull request merged**